libraries/provider_runit_service.rb

```diff
@@ -451,80 +452,92 @@ def get_current_env
-          @control_dir.group(new_resource.group)
-          @control_dir.mode(00755)
-          @control_dir
+          @control_dir ||=
+            begin
+              d = Chef::Resource::Directory.new(::File.join(sv_dir_name, 'control'), run_context)
+              d.owner(new_resource.owner)
+              d.group(new_resource.group)
+              d.mode(00755)
+              d
+            end
         end
 
         def control_signal_files
-          return @control_signal_files unless @control_signal_files.nil?
-          @control_signal_files = new_resource.control.map do |signal|
-            control_signal_file = Chef::Resource::Template.new(
-              ::File.join(sv_dir_name, 'control', signal),
-              run_context
-              )
-            control_signal_file.owner(new_resource.owner)
-            control_signal_file.group(new_resource.group)
-            control_signal_file.mode(00755)
-            control_signal_file.source("sv-#{new_resource.control_template_names[signal]}-#{signal}.erb")
-            control_signal_file.cookbook(template_cookbook)
-            control_signal_file.variables(:options => new_resource.options) if new_resource.options.respond_to?(:has_key?)
-            control_signal_file
-          end
-          @control_signal_files
+          @control_signal_files ||=
+            begin
+              new_resource.control.map do |signal|
+                t = Chef::Resource::Template.new(
+                  ::File.join(sv_dir_name, 'control', signal),
+                  run_context
+                )
+                t.owner(new_resource.owner)
+                t.group(new_resource.group)
+                t.mode(00755)
+                t.source("sv-#{new_resource.control_template_names[signal]}-#{signal}.erb")
+                t.cookbook(template_cookbook)
+                t.variables(:options => new_resource.options) if new_resource.options.respond_to?(:has_key?)
+                t
+              end
+            end
         end
 
         def lsb_init
-          return @lsb_init unless @lsb_init.nil?
-          initfile = ::File.join(new_resource.lsb_init_dir, new_resource.service_name)
-          if node['platform'] == 'debian'
-            ::File.unlink(initfile) if ::File.symlink?(initfile)
-            @lsb_init = Chef::Resource::Template.new(initfile, run_context)
-            @lsb_init.owner('root')
-            @lsb_init.group('root')
-            @lsb_init.mode(00755)
-            @lsb_init.cookbook('runit')
-            @lsb_init.source('init.d.erb')
-            @lsb_init.variables(:name => new_resource.service_name)
-          else
-            @lsb_init = Chef::Resource::Link.new(initfile, run_context)
-            @lsb_init.to(new_resource.sv_bin)
-          end
-          @lsb_init
+          @lsb_init ||=
+            begin
+              initfile = ::File.join(new_resource.lsb_init_dir, new_resource.service_name)
+              if node['platform'] == 'debian'
```

Comment:
We should probably ask about the underlying system's service provider since Debian will move to systemd in the next major version.